android/engine/src/main/java/org/smartregister/fhircore/engine/util/extension/PatientExtension.kt

```diff
@@ -211,3 +213,14 @@ fun Patient.extractOfficialIdentifier(): String? =
   if (this.hasIdentifier())
     this.identifier.firstOrNull { it.use == Identifier.IdentifierUse.OFFICIAL }?.value
   else null
+
+fun Patient.extractTypeViaDTreeMeta(): PatientType {
```

Comment:
This should be an extension function and should be reusable.

Reply:
its already as extension, though need to generalize and remove "Dtree"